traitsui/testing/tester/_ui_tester_registry/qt4/_interaction_helpers.py

```diff
@@ -205,6 +205,127 @@ def mouse_click_item_view(model, view, index, delay):
+    QTest.mouseDClick(
+        view.viewport(),
+        QtCore.Qt.LeftButton,
+        QtCore.Qt.NoModifier,
+        rect.center(),
+        delay=delay,
+    )
+
+
+def key_sequence_item_view(model, view, index, sequence, delay=0):
+    """ Perform mouse click on the given QAbstractItemModel (model) and
+    QAbstractItemView (view) with the given row and column.
+
+    Parameters
+    ----------
+    model : QAbstractItemModel
+        Model from which QModelIndex will be obtained
+    view : QAbstractItemView
+        View from which the widget identified by the index will be
+        found and key sequence be performed.
+    index : QModelIndex
+    sequence : str
+        Sequence of characters to be inserted to the widget identifed
+        by the row and column.
+
+    Raises
+    ------
+    Disabled
+        If the widget cannot be edited.
+    LookupError
+        If the index cannot be located.
+        Note that the index error provides more
+    """
+    check_q_model_index_valid(index)
+    widget = view.indexWidget(index)
+    if widget is None:
+        raise Disabled(
+            "No editable widget for item at row {!r} and column {!r}".format(
+                index.row(), index.column()
+            )
+        )
+    QTest.keyClicks(widget, sequence, delay=delay)
+
+
+def key_click_item_view(model, view, index, key, delay=0):
+    """ Perform key press on the given QAbstractItemModel (model) and
+    QAbstractItemView (view) with the given row and column.
+
+    Parameters
+    ----------
+    model : QAbstractItemModel
+        Model from which QModelIndex will be obtained
+    view : QAbstractItemView
+        View from which the widget identified by the index will be
+        found and key press be performed.
+    index : int
+    key : str
+        Key to be pressed.
+
+    Raises
+    ------
+    Disabled
+        If the widget cannot be edited.
+    LookupError
+        If the index cannot be located.
+        Note that the index error provides more
+    """
+    check_q_model_index_valid(index)
+    widget = view.indexWidget(index)
+    if widget is None:
+        raise Disabled(
+            "No editable widget for item at row {!r} and column {!r}".format(
+                index.row(), index.column()
+            )
+        )
+    key_click(widget, key=key, delay=delay)
+
+
+def get_display_text_item_view(model, view, index):
+    """ Return the textural representation for the given model, row and column.
+
+    Parameters
+    ----------
+    model : QAbstractItemModel
+        Model from which QModelIndex will be obtained
+    view : QAbstractItemView
+        View from which the widget identified by the index will be
+        found and key press be performed.
+    index : int
+
+    Raises
+    ------
+    LookupError
+        If the index cannot be located.
+        Note that the index error provides more
+    """
+    check_q_model_index_valid(index)
+    return model.data(index, QtCore.Qt.DisplayRole)
```

Comment:
is `data` the right thing here for displayed text? (look at Qt docs for `QAbstractItemModel`)

Reply:
Yes, `data` with `DisplayRole` should provide the text to display.